api/models/metadata.go

```diff
@@ -18,34 +18,47 @@ import (
 	"database/sql/driver"
 	"encoding/json"
 	"errors"
-	"fmt"
 
 	"github.com/gojek/merlin/mlp"
 	"github.com/gojek/merlin/utils"
 )
 
 const (
-	labelTeamName         = "gojek.com/team"
-	labelStreamName       = "gojek.com/stream"
-	labelAppName          = "gojek.com/app"
-	labelOrchestratorName = "gojek.com/orchestrator"
-	labelEnvironment      = "gojek.com/environment"
-	labelUsersPrefix      = "gojek.com/%s"
+	LabelAppName          = "app"
+	LabelComponent        = "component"
+	LabelEnvironment      = "environment"
+	LabelOrchestratorName = "orchestrator"
+	LabelStreamName       = "stream"
+	LabelTeamName         = "team"
+
+	ComponentBatchJob      = "batch-job"
+	ComponentImageBuilder  = "image-builder"
+	ComponentModelEndpoint = "model-endpoint"
+	ComponentModelVersion  = "model-version"
 )
 
 var reservedKeys = map[string]bool{
-	labelTeamName:         true,
-	labelStreamName:       true,
-	labelAppName:          true,
-	labelOrchestratorName: true,
-	labelEnvironment:      true,
+	LabelAppName:          true,
+	LabelComponent:        true,
+	LabelEnvironment:      true,
+	LabelOrchestratorName: true,
+	LabelStreamName:       true,
+	LabelTeamName:         true,
+}
+
+var prefix string
+
+// InitKubernetesLabeller builds a new KubernetesLabeller Singleton
+func InitKubernetesLabeller(p string) {
+	prefix = p
```

Comment:
can we do validation whether the prefix is valid or not? e.g "goto/gojek/" is not a valid prefix, if this value passed during init of merlin api, user won't be able to do anything

Reply:
Great suggestion. We added the validation for prefix.